src/Microsoft.Data/GroupBy.cs

```diff
@@ -64,21 +85,24 @@ public GroupBy(DataFrame dataFrame, int groupByColumnIndex, Dictionary<TKey, ICo
             _dataFrame = dataFrame ?? throw new ArgumentException(nameof(dataFrame));
         }
 
-        private delegate void ColumnDelegate(int columnIndex, long rowIndex, IEnumerable<long> rows, TKey key, bool firstGroup);
+        private delegate void ColumnDelegate(int columnIndex, long rowIndex, ICollection<long> rows, TKey key, bool firstGroup);
         private delegate void GroupByColumnDelegate(long rowNumber, TKey key);
-        private void EnumerateColumnsWithRows(GroupByColumnDelegate groupByColumnDelegate, ColumnDelegate columnDelegate)
+        private void EnumerateColumnsWithRows(GroupByColumnDelegate groupByColumnDelegate, ColumnDelegate columnDelegate, params string[] columnNames)
         {
             long rowNumber = 0;
             bool firstGroup = true;
             foreach (KeyValuePair<TKey, ICollection<long>> pairs in _keyToRowIndicesMap)
             {
                 groupByColumnDelegate(rowNumber, pairs.Key);
                 ICollection<long> rows = pairs.Value;
+                IEnumerable<string> columns = columnNames;
+                if (columnNames.Length == 0)
```

Comment:
`columnNames` can still be `null` because it gets passed in from public API that takes an array. Someone can call that public API passing in `null`.  ex. `.Max(null)`.